File: uri/calculosimples/calculosimples.pas
program foo;

uses 
    sysutils;

var
    codeA, codeB : integer;
    npecasA, npecasB : integer;
    valorA, valorB, resultado : double;

begin
    Readln (codeA, npecasA, valorA);
    Readln (codeB, npecasB, valorB );
    resultado := (npecasA * valorA ) + (npecasB * valorB);
    writeln ('VALOR A PAGAR: R$ ',Format('%.2f', [resultado]) );
end.

File: uri/calculosimples/calculosimples.ml
let calcular npecasA valorA npecasB valorB  = 
   (float_of_int npecasA *. valorA) +. (float_of_int npecasB *. valorB);;

let (codeA, npecasA, valorA) = 
   Scanf.scanf "%d  %d  %f\n" (fun codeA  npecasA  valorA -> (codeA, npecasA, valorA));;

let (codeB, npecasB, valorB) = 
   Scanf.scanf "%d  %d  %f\n" (fun codeB  npecasB  valorB -> (codeB, npecasB, valorB));;

let resultado = calcular npecasA valorA npecasB valorB;;

Printf.printf "VALOR A PAGAR: R$ %.2f\n" resultado;;